main/src/flix/runtime/fixpoint/ram/AssignStmt.java

```diff
@@ -0,0 +1,19 @@
+package flix.runtime.fixpoint.ram;
+
+import java.io.PrintStream;
+
+public class AssignStmt implements Stmt {
+    private TableName name;
+    private RelationExp relationExp;
+
+    public AssignStmt(TableName name, RelationExp relationExp) {
+        this.name = name;
```

Comment:
I suggest you add:

```
if (name == null) throw new IllegalArgumentException("'name' must be non-null");
```

and so forth, and for all constructors.

This is by far the best way to ensure that the RAM ast (which is constructed when the program is running!) remains safe.